components/MembersList/index.jsx

```diff
@@ -63,24 +66,28 @@ export const MembersList = () => {
 
   const onJoin = async () => {
     setJoinLoading(true);
-    const balance = await fetchVeolasBalance({ account });
-    if (Number(balance) === 0) {
-      notifyError(
-        <>
-          You must hold veOLAS to join.&nbsp;
-          <a
-            href="https://member.olas.network/"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Get veOLAS
-          </a>
-        </>,
-      );
-
+    try {
+      const balance = await fetchVeolasBalance({ account });
+      if (Number(balance) === 0) {
+        notifyError(
+          <>
+            You must hold veOLAS to join.&nbsp;
+            <a
+              href="https://member.olas.network/"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              Get veOLAS
+            </a>
+          </>,
+        );
+      } else {
+        signMessage();
+      }
+    } catch (error) {
+      notifyError('Error while fetching veOlas balance');
```

Comment:
For the user, the action they're trying to perform is Join, so we should just say "Join failed", and then log the error